rinex/src/lib.rs

```diff
@@ -1883,326 +1262,1322 @@ impl Rinex {
+        epochs.get(epochs.len() - 1).copied()
+    }
+
+    /// Returns Duration of (time spanned by) this RINEX
+    pub fn duration(&self) -> Option<Duration> {
+        let start = self.first_epoch()?;
+        let end = self.last_epoch()?;
+        Some(end - start)
+    }
+
+    /// Form a [`Timeseries`] iterator spanning [Self::duration]
+    /// with [Self::dominant_sample_rate] spacing
+    pub fn timeseries(&self) -> Option<TimeSeries> {
+        let start = self.first_epoch()?;
+        let end = self.last_epoch()?;
+        let dt = self.dominant_sample_rate()?;
+        Some(TimeSeries::inclusive(start, end, dt))
+    }
+
+    /// Returns sample rate used by the data receiver.
+    pub fn sample_rate(&self) -> Option<Duration> {
+        self.header.sampling_interval
+    }
+
+    /// Returns dominant sample rate
+    /// ```
+    /// use rinex::prelude::*;
+    /// let rnx = Rinex::from_file("../test_resources/MET/V2/abvi0010.15m")
+    ///     .unwrap();
+    /// assert_eq!(
+    ///     rnx.dominant_sample_rate(),
+    ///     Some(Duration::from_seconds(60.0)));
+    /// ```
+    pub fn dominant_sample_rate(&self) -> Option<Duration> {
+        if let Some(dominant) = self
+            .sampling_histogram()
+            .into_iter()
+            .max_by(|(_, x_pop), (_, y_pop)| x_pop.cmp(y_pop))
+        {
+            Some(dominant.0)
         } else {
-            // add special marker, ts: YYYYDDMM HHMMSS UTC
-            let now = hifitime::Epoch::now().expect("failed to retrieve system time");
-            let (y, m, d, hh, mm, ss, _) = now.to_gregorian_utc();
-            self.header.comments.push(format!(
-                "rustrnx-{:<20} FILE MERGE          {}{}{} {}{}{} {}",
-                env!("CARGO_PKG_VERSION"),
-                y + 1900,
-                m,
-                d,
-                hh,
-                mm,
-                ss,
-                now.time_scale
-            ));
-            // RINEX record merging
-            self.record.merge_mut(&rhs.record)?;
-            Ok(())
+            None
+        }
+    }
+
+    /// ```
+    /// use rinex::prelude::*;
+    /// use std::collections::HashMap;
+    /// let rinex = Rinex::from_file("../test_resources/NAV/V3/AMEL00NLD_R_20210010000_01D_MN.rnx")
+    ///     .unwrap();
+    /// let histogram: HashMap<_, _> = [
+    ///     (Duration::from_seconds(15.0*60.0), 1),
+    ///     (Duration::from_seconds(4.0*3600.0 + 45.0*60.0), 2),
+    ///     (Duration::from_seconds(25.0*60.0), 1),
+    ///     (Duration::from_seconds(5.0*3600.0 + 30.0*60.0), 1),
+    /// ].into_iter()
+    ///     .collect();
+    /// assert_eq!(rinex.sampling_histogram(), histogram);
+    /// ```
+    pub fn sampling_histogram(&self) -> HashMap<Duration, usize> {
+        let mut prev: Option<Epoch> = None;
+        let mut histogram: HashMap<Duration, usize> = HashMap::new();
+
+        for epoch in self.epoch() {
+            if let Some(prev) = prev {
```

Comment:
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.peekable might be better here.